sdk/core/core-lro/src/poller/models.ts

```diff
@@ -190,39 +214,22 @@ export interface SimplePollerLike<TState extends OperationState<TResult>, TResul
    * It returns a method that can be used to stop receiving updates on the given callback function.
    */
   onProgress(callback: (state: TState) => void): CancelOnProgress;
+
   /**
-   * Returns true if the poller has finished polling.
-   */
-  isDone(): boolean;
-  /**
-   * Stops the poller. After this, no manual or automated requests can be sent.
-   */
-  stopPolling(): void;
-  /**
-   * Returns true if the poller is stopped.
-   */
-  isStopped(): boolean;
-  /**
-   * Returns the state of the operation.
-   */
-  getOperationState(): TState;
-  /**
-   * Returns the result value of the operation,
-   * regardless of the state of the poller.
-   * It can return undefined or an incomplete form of the final TResult value
-   * depending on the implementation.
+   * Returns a promise that could be used for serialized version of the poller's operation
+   * by invoking the operation's serialize method.
    */
-  getResult(): TResult | undefined;
+  serialize(): Promise<string>;
+
   /**
-   * Returns a serialized version of the poller's operation
-   * by invoking the operation's toString method.
+   * Returns a promise that could be used to check if the poller has been submitted.
    */
-  toString(): string;
+  submitted(): Promise<void>;
 }
 
 /**
  * A state proxy that allows poller implementation to abstract away the operation
- * state. This is useful to implement `lroEngine` and `createPoller` in a modular
+ * state. This is useful to implement `createPoller` in a modular
  * way.
  */
 export interface StateProxy<TState, TResult> {
```

Comment:
Do we need the proxy stuff if we're removing the legacy implementations?

Reply:
Sure, removed.